packages/headless/src/features/commerce/product-templates/product-templates-helpers.ts

```diff
@@ -0,0 +1,108 @@
+import {isArray, isNullOrUndefined} from '@coveo/bueno';
+import {Product} from '../../../api/commerce/common/product';
+import {ProductTemplateCondition} from './product-templates-manager';
+
+/**
+ * Extracts a property from a product object.
+ * @param product (Product) The target product.
+ * @param property (string) The property to extract.
+ * @returns (unknown) The value of the specified property in the specified product, or null if the property does not exist.
+ */
+export const getProductProperty = (product: Product, property: string) => {
+  const anyProduct = product as unknown as Record<string, unknown>;
+
+  if (!isNullOrUndefined(anyProduct[property])) {
+    return anyProduct[property];
+  }
+
+  if (!isNullOrUndefined(product.additionalFields[property])) {
+    return product.additionalFields[property];
+  }
+
+  return null;
+};
+
+/**
+ * Creates a condition that verifies if the specified fields are defined.
+ * @param fieldNames (string[]) A list of fields that must be defined.
+ * @returns (ProductTemplateCondition) A function that takes a product and checks if every field in the specified list is defined.
+ */
+export const fieldsMustBeDefined = (
+  fieldNames: string[]
+): ProductTemplateCondition => {
+  return (product: Product) => {
+    return fieldNames.every(
+      (fieldName) => !isNullOrUndefined(getProductProperty(product, fieldName))
+    );
+  };
+};
+
+/**
+ * Creates a condition that verifies if the specified fields are not defined.
+ * @param fieldNames (string[]) A list of fields that must not be defined.
+ * @returns (ProductTemplateCondition) A function that takes a product and checks if every field in the specified list is not defined.
+ */
+export const fieldsMustNotBeDefined = (
+  fieldNames: string[]
+): ProductTemplateCondition => {
+  return (product: Product) => {
+    return fieldNames.every((fieldName) =>
+      isNullOrUndefined(getProductProperty(product, fieldName))
+    );
+  };
+};
+
+/**
+ * Creates a condition that verifies if a field's value contains any of the specified values.
+ * @param fieldName (string) The name of the field to check.
+ * @param valuesToMatch (string[]) A list of possible values to match.
+ * @returns (ProductTemplateCondition) A function that takes a product and checks if the value for the specified field matches any value in the specified list.
```

Comment:
```suggestion
 * @returns (ProductTemplateCondition) - A function that takes a Product as an argument and returns "true" if the value for the specified field is equal to any of the values in the specified list (case insensentive), and "false" otherwise.
```